logproof/src/linear_relation.rs

```diff
@@ -24,6 +24,22 @@ use crate::{
 };
 
 type MatrixPoly<Q> = Matrix<DensePolynomial<Q>>;
+type Bounds = Vec<u64>;
+
+impl Zero for Bounds {
+    // The empty vector could be seen as no bounds. Also follows the field
+    // properties.  Although realistically this would be indexed by the
+    // dimension d.
+    fn zero() -> Self {
+        Vec::new()
+    }
+}
```

Comment:
I don't love this implementation. I mostly wanted to use a matrix like container type so I can check that all the dimensions of `s` matched the dimensions of the bounds. We could add a new type for it, but I was attempting to reduce the number of new largely equivalent container types we had.

Reply:
We have determined this is ok for now :)